### PR TITLE
[Imt] Do not trigger the construction of TROOT when invoking IsImplicitMTEnabled()

### PR DESCRIPTION
**Pull request by dpiparo**
but rather keep track of the state as a boolean.

### CODE REVIEW

**Comment by pcanal** [comment]
The commit log and code comment seems at odd.  One talks about gROOT and the other talks about libIMT.

**Comment by dpiparo** [comment]
They do, but they are not at odd. libImt is presently the tool used via TROOT to get the status of the activation of ImplicitMT. That requires the load of libImt and the invocation of a function. This PR made of 1 commit simplifies the mechanism, avoiding to load libImt (and initialise the entire TROOT) in case only the activation status of ImplicitMT is required.

**Comment by pcanal** [comment]
Could the commit log be extended to include something explaining that?

Also could reduce the length of the first line of the commit log (see https://chris.beams.io/posts/git-commit/)

Thanks.

**Comment by dpiparo** [comment]
The message is now more verbose.

File: core/base/src/TROOT.cxx [hunk 1]
```diff
@@ -552,11 +569,7 @@ namespace Internal {
    Bool_t IsImplicitMTEnabled()
    {
 #ifdef R__USE_IMT
```
Comment (pcanal) [inline core/base/src/TROOT.cxx]: This #ifdef is now redundant :)

File: core/base/src/TROOT.cxx [hunk 2]
```diff
@@ -481,6 +481,21 @@ namespace Internal {
...
+   ////////////////////////////////////////////////////////////////////////////////
+   /// Keeps track of the status of ImplicitMT w/o resorting to the load of
+   /// libImt
+   static Bool_t &IsImplicitMTEnabledImpl()
+   {
+      static Bool_t isImplicitMTEnabled= kFALSE;
+#ifdef R__USE_IMT
+      return isImplicitMTEnabled;
+#else
+      return kFALSE;
```
Comment (pcanal) [inline core/base/src/TROOT.cxx]: Doesn't that return a reference to a temporary?

**Comment by dpiparo** [comment]
Both comments addressed. Thanks Philippe.

**Pull request merged**